mq.go

```diff
@@ -429,6 +447,7 @@ func (mq *mq) reconnect() {
 
 	mq.stopProducersAndConsumers()
 
+	atomic.StoreInt32(mq.state, ConnectionStateReconnecting)
```

Comment:
I understand the semantic of this status but I'm not sure it will be useful.
Is the difference between connecting and reconnecting states is important for you?